server/src/util.js

```diff
@@ -0,0 +1,43 @@
+const YAML = require('yamljs')
+const fs = require('fs')
+const path = require('path')
+
+function parseSpec(origFile, fileExt) {
+  if(fileExt === '.json') {
+    origFile = JSON.parse(origFile)
+    return origFile
+  }
+  else if(fileExt === '.yaml') {
+    origFile = YAML.parse(origFile)
+    return origFile
+  }
+  else {
+    return null
+  }
+}
+
+function sendResponse(err, result, file, syntax, toFile, response) {
+  const options = {syntax: 'yaml', order: 'openapi'}
+
+  if(err) {
+    response.status(500).send({'message': err})
+  }
+  else {
+    result = syntax === 'yaml' ? result.stringify(options) : result
+    if(toFile) {
+      const fileName = `${path.parse(file.originalname).name}.${syntax ? syntax : 'json'}` 
+      fs.writeFileSync(`./${fileName}`, syntax === 'yaml' ? result : JSON.stringify(result))
```

Comment:
@pcgamer1 did we explore using `res.send` to send a Buffer response, and setting a `content-type` header? This is instead of writing the output file to disk and sending that file as a response.

Reply:
Example to send Excel files using this method - https://stackoverflow.com/questions/51063253/how-to-send-buffer-data-in-express

Reply:
Thanks, using the buffer to send the response.